src/IPGenerate/IPGenerator.cpp

```diff
@@ -510,3 +512,21 @@ std::filesystem::path IPGenerator::GetBuildDir(IPInstance* instance) const {
   }
   return dir;
 }
+
+// This will return the path to this instance's cached json file
+std::filesystem::path IPGenerator::GetCachePath(IPInstance* instance) const {
+  std::filesystem::path dir{};
+
+  Compiler* compiler = GlobalSession->GetCompiler();
+
+  ProjectManager* projManager = nullptr;
```

Comment:
I don't thing you need this.

Reply:
I was getting a compiler error when doing just GetCompiler() for some reason, the const qualifier was my only guess.

Reply:
if (compiler && compiler->ProjManager()) should work. If not, please fix

Reply:
Oh i see what you mean now, i can just use m_compiler w/o a getter, i didn't realize the parent class already had its own ptr. Good catch! Fixed.